src/sample.jl

```diff
@@ -122,40 +144,58 @@ function mcmcsample(
         end
 
         # Obtain the initial sample and state.
-        sample, state = step(rng, model, sampler; kwargs...)
+        sample, state = if num_warmup > 0
+            step_warmup(rng, model, sampler; kwargs...)
+        else
+            step(rng, model, sampler; kwargs...)
+        end
+
+        # Update the progress bar.
+        itotal = 1
+        if progress && itotal >= next_update
+            ProgressLogging.@logprogress itotal / Ntotal
+            next_update = itotal + threshold
+        end
 
         # Discard initial samples.
-        for i in 1:discard_initial
-            # Update the progress bar.
-            if progress && i >= next_update
-                ProgressLogging.@logprogress i / Ntotal
-                next_update = i + threshold
+        for j in 1:discard_initial
+            # Obtain the next sample and state.
+            sample, state = if j ≤ num_warmup
```

Comment:
This should be
```suggestion
            sample, state = if j ≤ discard_num_warmup
```
shouldn't it?

Maybe it could even be split into two sequential for loops?

Reply:
> This should be

I think _technically_ it doesn't matter, right? Since we have either
1. `discard_num_warmup = discard_initial` when `num_warmup >= discard_initial`, or
2. `discard_num_warmup = num_warmup` when `num_warmup < discard_initial`.

In both of those cases we get the same behavior in the above.

_But_ I think for readability's sake, I agree we should make the change! Just pointing out it shouldn't been a cause of a bug.

> Maybe it could even be split into two sequential for loops?

Wait what, wasn't that what I had before? :confused:

Reply:
> Wait what, wasn't that what I had before? confused

Really? I think you used a different logic initially but maybe I misremember :smile: In any case, I guess it does not matter.

Reply:
You mean

```julia
for i = 1:discard_num_warmup
    # ...
end

for i = discard_num_warmup + 1:discard_initial
    # ...
end
```

?

Because you're probably right, I don't think I ever did this exactly :grimacing: 

I'm preferential to the current code for readability's sake because it means the discard stepping is looks the same as the proper stepping, code-wise.